plugins/scanners/fossid/src/main/kotlin/FossId.kt

```diff
@@ -922,4 +927,42 @@ class FossId internal constructor(
             mapOf(SCAN_CODE_KEY to scanCode, SCAN_ID_KEY to scanId, SERVER_URL_KEY to config.serverUrl)
         )
     }
+
+    /**
+     * Mark all the files having a snippet choice as identified, only if they have no non-chosen source location
+     * remaining.
+     */
+    private fun markFilesWithChosenSnippetsAsIdentified(
+        scanCode: String,
+        snippetChoices: List<SnippetChoice> = emptyList(),
+        snippetFinding: Set<SnippetFinding>,
+        listPendingFiles: List<String>
+    ) {
+        runBlocking(Dispatchers.IO) {
+            val candidatePathsToMark = snippetChoices.groupBy({ it.given.sourceLocation.path }) {
+                it.choice.reason
+            }
+
+            val requests = mutableListOf<Deferred<Any>>()
+
+            candidatePathsToMark.forEach { (path, reasons) ->
+                val hasNonChosenLocations = snippetFinding.any { path == it.sourceLocation.path }
```

Comment:
Maybe use `none` instead of `any` (and rename the variable accordingly) to avoid the double-negation in the condition below?